### PR TITLE
EG-782, moved AuditInfo generation into service

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/services/ReviewService.scala [hunk 1]
```diff
@@ -22,14 +22,15 @@ import models.errors._
...
-
+import models.ocelot.Process
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
 class ReviewService @Inject() (publishedService: PublishedService, repository: ApprovalRepository, reviewRepository: ApprovalProcessReviewRepository) {
 
   val logger: Logger = Logger(this.getClass)
+  val br: RequestOutcome[AuditInfo] = Left(Errors(BadRequestError))
```
Comment (LesWilson) [inline app/services/ReviewService.scala:33]: Can't see this being used

File: app/services/ReviewService.scala [hunk 2]
```diff
@@ -85,12 +92,16 @@ class ReviewService @Inject() (publishedService: PublishedService, repository: A
...
+  def factCheckComplete(id: String, info: ApprovalProcessStatusChange): Future[RequestOutcome[AuditInfo]] =
     checkProcessInCorrectStateForCompletion(id, ReviewTypeFactCheck) flatMap {
-      case Right(approvalProcess) =>
-        reviewRepository.updateReview(id, approvalProcess.version, ReviewTypeFactCheck, info.userId, info.status) flatMap {
+      case Right(ap) =>
+        reviewRepository.updateReview(id, ap.version, ReviewTypeFactCheck, info.userId, info.status) flatMap {
           case Right(_) => changeStatus(id, info.status, info.userId, ReviewTypeFactCheck) map {
-            case Right(_) => Right(approvalProcess)
+            case Right(_) => ap.process.validate[Process].fold(
+                  _ => Left(Errors(BadRequestError)): RequestOutcome[AuditInfo],
```
Comment (LesWilson) [inline app/services/ReviewService.scala:102]: whitespace at end of line

File: app/services/ReviewService.scala [hunk 3]
```diff
@@ -68,11 +69,17 @@ class ReviewService @Inject() (publishedService: PublishedService, repository: A
...
-      case Right(approvalProcess) =>
-        reviewRepository.updateReview(id, approvalProcess.version, ReviewType2i, info.userId, info.status) flatMap {
+      case Right(ap) =>
+        reviewRepository.updateReview(id, ap.version, ReviewType2i, info.userId, info.status) flatMap {
           case Right(()) =>
             changeStatus(id, info.status, info.userId, ReviewType2i) flatMap {
-              case Right(_) => publishIfRequired(approvalProcess)
+              case Right(_) => publishIfRequired(ap).map{
+                case Right(_) => ap.process.validate[Process].fold(
+                  _ => Left(Errors(BadRequestError)): RequestOutcome[AuditInfo],
```
Comment (LesWilson) [inline app/services/ReviewService.scala:78]: whitespace at end of line

File: app/services/ReviewService.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/services/ReviewService.scala:33]: oops, want intended to be left

File: app/services/ReviewService.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/services/ReviewService.scala:102]: will remove

File: app/services/ReviewService.scala (hunk 3 above)
Comment (cmrhmrc) [reply app/services/ReviewService.scala:78]: will remove

**Pull request merged**